### PR TITLE
Remove unrequired assertion for invalid markup (fixes #305)

### LINKED ISSUE #305: Crash when parsing an argument value that is `\r\n`
Hi,

I was trying to fuzz ammonia to see if there was anything that would make it crash and found this string `<a a=\r\n` that makes html5ever panic.

Here is the test example:

```rust
test!(fuzz, "<a a=\r\n", "");
```

The panic/backtrace:
```
thread '<unnamed>' panicked at 'assertion failed: c.is_some()', /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:555:8
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:611
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:572
   6: <html5ever::tokenizer::Tokenizer<Sink>>::discard_char
             at ./<panic macros>:3
   7: <html5ever::tokenizer::Tokenizer<Sink>>::step
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:570
   8: <html5ever::tokenizer::Tokenizer<Sink>>::run
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:362
   9: <html5ever::tokenizer::Tokenizer<Sink>>::feed
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/tokenizer/mod.rs:220
  10: <html5ever::driver::Parser<Sink> as tendril::stream::TendrilSink<tendril::fmt::UTF8>>::process
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/html5ever-0.19.0/src/driver.rs:88
  11: <tendril::stream::Utf8LossyDecoder<Sink, A> as tendril::stream::TendrilSink<tendril::fmt::Bytes, A>>::process
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/tendril-0.4.0/src/stream.rs:179
  12: tendril::stream::TendrilSink::read_from
             at /home/eijebong/.cargo/registry/src/github.com-1ecc6299db9ec823/tendril-0.4.0/src/stream.rs:79
  13: rust_fuzzer_test_input
             at fuzz/fuzz_targets/fuzz_target_1.rs:17
  14: libfuzzer_sys::test_input_wrap::{{closure}}
             at /home/eijebong/.cargo/git/checkouts/libfuzzer-sys-e07fde05820d7bc6/67f7399/src/lib.rs:11
  15: std::panicking::try::do_call
             at /checkout/src/libstd/panicking.rs:480
  16: <unknown>
             at /checkout/src/libpanic_abort/lib.rs:40
```

### PR DESCRIPTION
**Pull request by mozfreddyb**

### CODE REVIEW

**Comment by emilio** [comment]
Any chance to add a test for this?

r? @nox

**Comment by nox** [comment]
A test would be nice indeed.

**Comment by mozfreddyb** [comment]
Where would you suggest I put a test?
The html5lib-tests folder looks like its something from another project

File: html5ever/tests/issue-305.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#[cfg(test)]
+fn main() {
+    let t = tokenizer::Tokenizer::new();
+    let r = t.feed("<a a=\r\n");
+    t.end();
+}
+
+
+
+// tokenize this test!(fuzz, "<a a=\r\n", "");
```
Comment (KiChjang) [inline html5ever/tests/issue-305.rs]: nit: missing newline at the end of this file

**Comment by Ygg01** [comment]
@mozfreddyb 
> Where would you suggest I put a test?

A fair question. I think the proper place is somewhat surprisingly in [serializer.rs](https://github.com/servo/html5ever/blob/0716de2b4b85c80167e83933431e0ee87898f4bd/html5ever/tests/serializer.rs#L191). Others will correct me if I'm mistaken.

Try adding the `test!(fuzz, "<a a=\r\n",  "");`. 

There is macro `test!($name, $input, $output)` where `$name` is name of test, `$input` is input string  and $output is output string. If no output is defined it's assumed `$output = $input`

**Comment by mozfreddyb** [comment]
Thanks @Ygg01. Your comment arrived after I had a brief chat with @nox on IRC. The `test!` macro approach seems interesting.

(I'll hope I will be able to take a look later today, but there's a risk of me dropping this pull request as I'm going on leave very soon.)

**Comment by mozfreddyb** [comment]
OK, I'm back.
@Ygg01 can you take a look?

**Comment by mozfreddyb** [comment]
@KiChjang Can you review? This should be complete.

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Pull request merged**